engine/src/flutter/lib/web_ui/lib/painting.dart

```diff
@@ -642,39 +642,49 @@ Future<Codec> instantiateImageCodecFromBuffer(
   int? targetWidth,
   int? targetHeight,
   bool allowUpscaling = true,
-}) => engine.renderer.instantiateImageCodec(
-  buffer._list!,
-  targetWidth: targetWidth,
-  targetHeight: targetHeight,
-  allowUpscaling: allowUpscaling,
-);
+}) {
+  try {
+    return engine.renderer.instantiateImageCodec(
+      buffer._list!,
+      targetWidth: targetWidth,
+      targetHeight: targetHeight,
+      allowUpscaling: allowUpscaling,
+    );
+  } finally {
+    buffer.dispose();
```

Comment:
This look wrong for me. An object is expected to be disposed by its creator. And buffer is not created here.

Reply:
I fixed it according to the issue and dartdoc, but it looks good to me too that "An object is expected to be disposed by its creator.".

Reply:
I have added a fix. Perhaps adding a mechanism to track `ImmutableBuffer` and `ImageDescriptor` leaks would make it easier to find them.

This fix conflicts with https://github.com/flutter/flutter/pull/159945 and I will rebase later 👍